docs/developers/frontend-shared.rst

```diff
@@ -0,0 +1,67 @@
+=========================
+Frontend Shared Package
+=========================
+
+The frontend-shared package is located in a sub directory called ``./frontend-shared`` inside this
+repository. The purpose of this package is to serve as a common library for shared utilities,
+components and styles that span across our frontend applications.
+
+The long term goal is to move this package into its own repository, but in the short term it
+lives in the client as a convenience to leverage the existing tooling. This also gives developers
+the benefit of working with just one repository when making changes that mostly affect the client
+in the first place.
+
+How it works
+=========================
+
+Frontend-shared works by creating a symlink via ``yarn link`` between the working dir
+``./frontend-shared/`` and the installed dir ``node_modules/@hypothesis/frontend-shared/``.
+
+Client setup
+=========================
+
+To setup frontend-shared in the client, simply run
+
+``$ make dev``
+
+Internally, that will trigger the frontend-shared setup which transpiles all of the jsx and
+outputs that result to ``frontend-shared/lib`` folder. The next thing it does is creates a linked
+package in the client ``node_modules/@hypothesis`` folder which points to ``frontend-shared/lib``.
+When running `make dev` this pipeline is automated and will trigger on file changes. These
+steps can also be run directly too via
+
+``$ gulp setup-frontend-shared``
+
+Usage in other repositories
+============================
+If you want to use the frontend-shared package another repository which imports it, such as LMS, but
+you want to see the modified changes in that repo before committing them, then you can create a
+link in that repo. If other words, you want to test uncommitted changes in frontend-shared in a
+second repository. First make sure that the link was set up by running
+
+``$ gulp setup-frontend-shared``
+
+inside client repo. Then run
+
+``$ yarn link @hypothesis/frontend-shared``
+
+inside the target consumption repo. With this work flow, you'll have to manually run
+
+``$ gulp build-frontend-shared-js``
+
+to update the ``/lib`` in frontend-shared when any changes are made to it or have the client
+running locally via ``made dev`` so changes are automatically built.
```

Comment:
```suggestion
If you have modified the ``frontend-shared`` package locally and would like to preview changes in another Hypothesis application that imports the package—that is, if you want to test uncommitted changes in ``frontend-shared`` in another local repository before publishing changes—follow these steps:

#. In this repository's directory:

    ``$ gulp setup-frontend-shared``
    
#. In the target repository:

    ``$ yarn link @hypothesis/frontend-shared`` 

Note: the package will need to be rebuilt after any changes are made to it before those changes can be seen in the consuming repository. This can be done by:

* Running ``make dev`` in this repository, which will automatically re-build on any file changes, or
* ``$ gulp build-frontend-shared-js`` in this repository to re-build manually

```

Reply:
Granted, this concept is a "little hard to explain"...whew!